src/App/ExpressionParser.h

```diff
@@ -101,7 +106,7 @@ class  AppExport UnitExpression : public Expression {
 
 private:
     Base::Quantity quantity;
-    std::string unitStr; /**< The unit string from the original parsed string */
+    const char *unitStr; /**< The unit string from the original parsed string */
```

Comment:
Why has this type been changed?

I can see that the UnitExpression constructor has been updated such that it takes 2 rather than 3 arguments, with `unitstr` defaulting to 0 in a protected constructor when needed but...why?

What is the purpose of this change? It seems to obfuscate and make things more difficult than they need to be.

Reply:
The purpose of commit 700c41af is to concentrate all unit string in one place as constant string, for both parsing and printing. Any unknown unit will be interpreted as a VariableExpression. So there is little reason to use std::string here.